src/components/BookForm.js

```diff
@@ -2,26 +2,37 @@ import React, { useState } from 'react';
 import './styles/BookForm.css';
 import { nanoid } from '@reduxjs/toolkit';
 import { useDispatch } from 'react-redux';
-import { addBook } from '../redux/books/booksSlice';
+import { addBook, postBook } from '../redux/books/booksSlice';
 
 function BookForm() {
   const [title, setTitle] = useState('');
   const [author, setAuthor] = useState('');
+  const [error, setError] = useState('');
+
   const dispatch = useDispatch();
 
   const handleAdd = () => {
     // Check if title and author are not empty
     if (title.trim() !== '' && author.trim() !== '') {
       dispatch(
+        postBook({
+          item_id: nanoid(),
+          title,
+          author,
+          category: 'unknown', // typo fixed
+        }),
         addBook({
           item_id: nanoid(),
           title,
           author,
           category: 'unknown', // typo fixed
         }),
       );
+      setError('');
       setTitle('');
       setAuthor('');
```

Comment:
- [ ] After adding a book it does not appear on the page immediately. I have to reload the browser for the newly added book to show up. Kindly note that when a book is added it should be displayed without having to reload the page. Kindly fix that.